index.js

```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+const Express = require('express')
```

Comment:
I don't understand why express dependency has been added.
In fastify core there is an array of Middlewares executed with `reply.context._middie.run(request.raw, reply.res, reply)`
Shouldn't this be enough in order to release fastify v2 with this plugin inside? (and then detach it in v3)
Maybe we want to support more express features like routers as @StarpTech was pointing out?
In case, would we like to be 100% compliant with the express middlewares?

Reply:
As you can see in https://github.com/fastify/fastify/issues/1503, the idea is to get rid of Middie and the `use` API in v3, since it not adds full compatibility with Express middlewares and offer this as an external plugin to have a complete compatibility layer, to help the developer in the transition from Express to Fastify.

Reply:
IMHO I would keep a `fastify-middleware` plugin based on middie. Middie is actually a good module ;).

Reply:
> IMHO I would keep a fastify-middleware plugin based on middie. Middie is actually a good module ;).

We can do that, but I think it is an effort that is more trouble than it's worth. At the end, we should basically replicate the entire Express behavior (which is very bug prone) to offer a feature that we want the developers not to use because we want them to use our hooks instead.

Reply:
We should do that for backward compatibility with previous fastify releases anyway.

Reply:
> We should do that for backward compatibility with previous fastify releases anyway.

Maybe I'm missing something, why this would break the backward compatibility?

Reply:
it’s a complete different codebase. Also, middie is fast.

Reply:
> it’s a complete different codebase. 

yes, but Middies aims to be compatible with Express itself, so this would even increase the compatibility ;)

> Also, middie is fast.

100% agree, but see https://github.com/fastify/fastify-express/pull/1#discussion_r264127050 :D

Reply:
I don’t think this is worth discussing in this PR. Keep going. Exposing middie as a plugin is a 30 min job and I’ll take care of it.